content/blog/scaling-nlp-indexing-pipelines-with-keda-and-haystack-part-1/index.md

```diff
@@ -0,0 +1,437 @@
+```yaml
+# docker-compose.yaml  
+# link to file: https://github.com/ArzelaAscoIi/haystack-keda-indexing/blob/main/docker-compose.yaml  
+...  
+consumer:  
+    build:  
+      context: .  
+      dockerfile: Dockerfile  
+    environment:  
+      - AWS_ENDPOINT=http://localstack:4566  
+      - AWS_REGION=eu-central-1  
+      - AWS_ACCESS_KEY_ID=test  
+      - AWS_SECRET_ACCESS_KEY=test
+```
+
+We are now ready to test our application by uploading files and checking the logs for resulting documents.
+
+## Testing with Docker Compose
+
+To start our Docker Compose stack, we use the command  `docker-compose up`. Additionally, we provide a code snippet for file uploading.
+
+```python
+# upload.py  
+# link to file: <https://github.com/ArzelaAscoIi/haystack-keda-indexing/blob/main/upload.py>  
+  
+aws_service = AWSService(SQS_QUEUE, S3_BUCKET, LOCAL_DOWNLOAD_DIR)  
+aws_service.upload_file(Path("./data/test.txt"))
+```
+
+If we run python3  `upload.py`  upload.py now, we will see the following output in our Docker console.
+
+```bash
+haystack-keda-indexing-localstack-1  | 2023-04-22T09:55:44.981  INFO --- [   asgi_gw_1] localstack.request.aws     : AWS s3.GetObject => 200  
+haystack-keda-indexing-localstack-1  | 2023-04-22T09:55:45.007  INFO --- [   asgi_gw_0] localstack.request.aws     : AWS sqs.DeleteMessage => 200  
+haystack-keda-indexing-consumer-1    | 2023-04-22 09:55:45 [info     ] Found files                    files=[PosixPath('/tmp/test.txt')]  
+Converting files: 100%|██████████| 1/1 [00:00<00:00, 81.28it/s]  
+Preprocessing: 100%|██████████| 1/1 [00:00<00:00, 35.10docs/s]  
+Batches: 100%|██████████| 1/1 [00:03<00:00,  3.18s/it]  
+haystack-keda-indexing-consumer-1    | 2023-04-22 09:55:48 [info     ] Processed files                documents={'documents': [<Document: {'content': 'this is text', 'content_type': 'text', 'score': None, 'meta': {'_split_id': 0}, 'id_hash_keys': ['content'], 'embedding': '<embedding of shape (768,)>', 'id': '46ec22c7eafaea7c43eef7d996fd04ce'}>], 'file_paths': [PosixPath('/tmp/test.txt')], 'root_node': 'File', 'params': {}, 'node_id': 'Retriever'}  
+haystack-keda-indexing-localstack-1  | 2023-04-22T09:55:48.327  INFO --- [   asgi_gw_0] localstack.request.aws     : AWS sqs.GetQueueUrl => 200
+```
+
+We have a stateless service that processes indexing requests asynchronously and can operate in Docker Compose. We can now deploy this service to Kubernetes, configure KEDA, and scale it.
+
+## Conclusion and next steps
+
+This article explains how to create a stateless consumer application that turns files into documents. We learned that we can use this application with different models, preprocessors, and file types which are defined in a single configuration file using Haystack. In order to ship this application to a production-ready deployment, we will deploy this service on Kubernetes and define custom autoscaling in  [the next chapter](scaling-nlp-indexing-pipelines-with-keda-and-haystack-part-2).
```

Comment:
I think we need to give the full link to the other blog here: https://haystack.deepset.ai/blog/scaling-nlp-indexing-pipelines-with-keda-and-haystack-part-2

Reply:
no, this is ok :)